secp256k1-sys/src/lib.rs

```diff
@@ -1084,6 +1171,117 @@ impl <T: CPtr> CPtr for Option<T> {
+    pub fn new(scan_pubkey: &PublicKey,  spend_pubkey: &PublicKey, index: usize) -> Self {
+        Self {
+            scan_pubkey: scan_pubkey.clone(),
+            spend_pubkey: spend_pubkey.clone(),
+            index
+        }
+    }
+}
+
+impl core::fmt::Debug for SilentpaymentsRecipient {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        f.debug_struct("SilentpaymentsRecipient")
+            .field("scan_pubkey", &self.scan_pubkey)
+            .field("spend_pubkey", &self.spend_pubkey)
+            .field("index", &self.index)
+            .finish()
+    }
+}
+
+#[cfg(not(secp256k1_fuzz))]
+impl PartialEq for SilentpaymentsRecipient {
+    fn eq(&self, other: &Self) -> bool {
+        self.scan_pubkey == other.scan_pubkey &&
+        self.spend_pubkey == other.spend_pubkey &&
+        self.index == other.index
+    }
+}
+
+#[cfg(not(secp256k1_fuzz))]
+impl Eq for SilentpaymentsRecipient {}
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct SilentpaymentsFoundOutput {
+    pub output: XOnlyPublicKey,
+    tweak: [u8; 32],
+    found_with_label: bool,
+    label: PublicKey,
+}
+
+impl SilentpaymentsFoundOutput {
+
+    pub fn empty() -> Self {
+        Self {
+            output: unsafe { XOnlyPublicKey::new() },
+            tweak: [0; 32],
+            found_with_label: false,
+            label: unsafe { PublicKey::new() },
+        }
+    }
+
+    pub fn new(output: &XOnlyPublicKey, tweak: &[u8; 32],  found_with_label: bool, label: &PublicKey) -> Self {
+        Self {
+            output: output.clone(),
+            tweak: tweak.clone(),
+            found_with_label,
+            label: label.clone(),
+        }
+    }
+}
+
+#[cfg(not(secp256k1_fuzz))]
+impl PartialEq for SilentpaymentsFoundOutput {
+    fn eq(&self, other: &Self) -> bool {
+        self.tweak == other.tweak &&
+        self.found_with_label == other.found_with_label &&
+        self.label == other.label
+    }
+}
+
+#[cfg(not(secp256k1_fuzz))]
+impl Eq for SilentpaymentsFoundOutput {}
+
+impl core::fmt::Debug for SilentpaymentsFoundOutput {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        f.debug_struct("SilentpaymentsFoundOutput")
+            .field("output", &self.output)
+            .field("tweak", &self.tweak)
+            .field("found_with_label", &self.found_with_label)
+            .field("label", &self.label)
+            .finish()
+    }
+}
+
+#[repr(C)]
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct SilentpaymentsPublicData([u8; 98]);
+
+impl SilentpaymentsPublicData {
+    pub fn from_array(arr: [u8; 98]) -> Self {
+        SilentpaymentsPublicData(arr)
+    }
+    pub fn to_array(self) -> [u8; 98] {
+        self.0
+    }
+}
+
+impl_array_newtype!(SilentpaymentsPublicData, u8, 98);
+impl_raw_debug!(SilentpaymentsPublicData);
+
+pub type SilentpaymentsLabelLookupFunction =  unsafe extern "C" fn(*const c_uchar, *const c_void) -> *const c_uchar;
```

Comment:
```suggestion
pub type SilentpaymentsLabelLookupFunction = Option<unsafe extern "C" fn(*const c_uchar, *const c_void) -> *const c_uchar>;
```

This should allow calling `secp256k1_silentpayments_recipient_scan_outputs` with a NULL `label_lookup`.
Uses in this pull request will have to be updated by wrapping the function pointer in `Some(...)`